python/marvin/contrib/vacs/hi.py

```diff
@@ -66,10 +122,37 @@ def get_target(self, parent_object):
         if not self.file_exists(self.summary_file):
             self.summary_file = self.download_vac('mangahisum', path_params=self.path_params)
 
+        #find all entries in summary file with this plate-ifu. I need the full summary file here. Find best entry between GBT/ALFALFA based on dept and confusion. Then update self.path_params['program'] with alfalfa or gbt.
+        
+        #summary = VACs.HI.get_table()
+        summary = VACs().HI.data[1].data
+        galinfo = summary[summary['plateifu'] == plateifu]
+        if len(galinfo) == 1:
+            if galinfo['session']=='ALFALFA':
+                program = 'alfalfa'
+            else:
+                program = 'gbt'
+        else:
+            par1={'program':'gbt','snr':0.,'rms':galinfo[0]['rms'],'conf_prob':galinfo[0]['conf_prob']}
+            par2={'program':'gbt','snr':0.,'rms':galinfo[1]['rms'],'conf_prob':galinfo[1]['conf_prob']}
+            if galinfo[0]['session']=='ALFALFA':
+                par1['program']='alfalfa'
+            if galinfo[1]['session']=='ALFALFA':
+                par2['program']='alfalfa'
+            if galinfo[0]['fhi'] > 0:
+                par1['snr'] = galinfo[0]['fhi']/galinfo[0]['efhi']
+            if galinfo[1]['fhi'] > 0:
+                par2['snr'] = galinfo[1]['fhi']/galinfo[1]['efhi']
+
+            program = choose_program(par1,par2)
```

Comment:
Does this logic work as expected for loading data from DR15 or DR16?  Since Marvin needs to be backwards compatible, I wonder if it's worth moving this into a standalone function, and putting the call inside a `if parent_object.release == "DR17"` condition.

Reply:
right, it's not compatible. I can make the modification you suggest